pulling fix java and completed sprintg context task 1

Vlagogaik opened the pull request.

Vlagogaik committed:
pulling

Vlagogaik committed:
Finish IO

Vlagogaik committed:
Finish IO

Vlagogaik committed:
switch to "org.oleg"

Vlagogaik committed:
Fixed problems

Vlagogaik committed:
Fixed problems

Vlagogaik committed:
Start to try spring contex

Vlagogaik committed:
Start to try spring contex

Vlagogaik committed:
Commit first task

Vlagogaik committed:
Commit first task

Vlagogaik committed:
Commit first task

Vlagogaik committed:
Delete spring/.mvn/wrapper directory

Vlagogaik committed:
Delete SbT-Pre/src/main/java/org/springs/contexts directory

Vlagogaik committed:
Delete SbT-Pre/src/test/java/spring directory

Vlagogaik committed:
Update StartJavaTest.java

Vlagogaik committed:
Complete task 1

Vlagogaik committed:
Merge remote-tracking branch 'origin/newOleg' into OlegUltima11

XLelarX commented on spring/src/main/java/org/context/task1/bean/AppConfing.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Bean
+    public Human person() {
+        return new Human(parrot1(), parrot2(), cat(), dog());
```
Инъекция зависимостей должна быть через аргументы метода

XLelarX commented on spring/src/main/java/org/context/task1/component/Cat.java:
```diff
@@ -0,0 +1,21 @@
+    private String name;
+
+    @Override
+    public String toString() {
```
Заюзай @Data просто

XLelarX commented on spring/src/main/java/org/context/task1/component/Human.java:
```diff
@@ -0,0 +1,24 @@
+@Setter
+@Component
+public class Human {
+    private org.context.task1.component.Parrot parrot1;
+    private org.context.task1.component.Parrot parrot2;
+    private org.context.task1.component.Cat cat;
+    private org.context.task1.component.Dog dog;
```
пакеты явно можно не указывать

XLelarX commented on spring/src/main/java/org/context/task1/component/Main.java:
```diff
@@ -0,0 +1,16 @@
+public class Main {
+    public static void main(String[] args) {
+        SpringApplication.run(Main.class, args);
+        ApplicationContext context = new AnnotationConfigApplicationContext("org.context.task1.component");
```
Если ты стартуешь через SpringBoot, то юзай `@ComponentScan`, а не ручное просечивание в контекст

Vlagogaik committed:
fixed task 1:
1. spring/src/main/java/org/context/task1/bean/AppConfing.java. Fixed Human.
2.spring/src/main/java/org/context/task1/component/Cat.java. Fixed Cat, toString has changed to @Data.
3. spring/src/main/java/org/context/task1/component/Human.java. Delete useless code.
4. spring/src/main/java/org/context/task1/component/Main.java. Using @ComponentScan, in AppConfing.java, but i am using @SpringBootApplication and it already has this feature enabled.

Vlagogaik committed:
Complete tast2
